slurm/src/slurmcc/cli.py

```diff
@@ -293,7 +321,112 @@ def resume_fail(
         Equivalent to SuspendFailProgram, shutsdown nodes
         """
         node_mgr = self._node_mgr(config, self._driver(config))
-        self._shutdown(node_list=node_list, node_mgr=node_mgr)
+        self._shutdown(config, node_list=node_list, node_mgr=node_mgr)
+
+    def return_to_idle_parser(self, parser: ArgumentParser) -> None:
+        parser.set_defaults(read_only=False)
+        parser.add_argument("--terminate-zombie-nodes", action="store_true", default=False)
+
+    def return_to_idle(
+        self, config: Dict, terminate_zombie_nodes: bool = False
+    ) -> None:
+        """
+        Nodes that fail to resume in ResumeTimeout seconds will be left
+        in a down~ state - i.e. down and powered_down. It is also possible
+        the nodes will be in a drained~ state, if the node was drained during
+        resume. This command will set those nodes to idle~.
+
+        The one exception is for nodes that have KeepAlive set in CycleCloud.
+        Those nodes will be left as down~ and will be logged. When the user
+        unclicks the KeepAlive, the node can be automatically shutdown if --terminate-zombie-nodes
+        is set, or config["return-to-idle"]["terminate-zombie-nodes"] is true.
+        """
+        if not slutil.is_autoscale_enabled():
+            return
+
+        # this is always run as root, so bump up the loglevel to info
+        stream_handlers = [
+            x
+            for x in logging.getLogger().handlers
+            if isinstance(x, logging.StreamHandler)
+        ]
+        for sh in stream_handlers:
+            sh.setLevel(logging.INFO)
+
+        node_mgr = self._node_mgr(config)
+        ccnodes = node_mgr.get_nodes()
+        ccnodes_by_name = hpcutil.partition_single(ccnodes, lambda node: node.name)
+
+        snodes = slutil.show_nodes()
+        if terminate_zombie_nodes:
+            if "return_to_idle" not in config:
+                config["return_to_idle"] = {}
+            config["return_to_idle"]["terminate-zombie-nodes"] = True
+            
+        SlurmCLI._return_to_idle(config, snodes, ccnodes_by_name, scontrol, node_mgr)
+
+    @staticmethod
```

Comment:
As we discussed we can simplify and refactor  this code a little bit by only returning to idle nodes that don't exist. And taking action (shutdown, or mark down) nodes that need to be shut down.